#! /bin/sh /usr/share/dpatch/dpatch-run
## 79_fix_ls_test_for_sticky.dpatch by Michael Stone <mstone@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix test which assumes file is not world writable

@DPATCH@
diff -urNad coreutils-7.5~/tests/misc/ls-misc coreutils-7.5/tests/misc/ls-misc
--- coreutils-7.5~/tests/misc/ls-misc	2009-08-15 11:25:32.000000000 -0400
+++ coreutils-7.5/tests/misc/ls-misc	2009-09-04 16:23:05.165410930 -0400
@@ -58,7 +58,7 @@
   my $test = shell_quote "$ENV{abs_top_builddir}/src/test";
   system (qq(touch setuid && chmod u+s setuid && $test -u setuid &&
 	   touch setgid && chmod g+s setgid && $test -g setgid &&
-	   mkdir sticky && chmod +t sticky  && $test -k sticky &&
+	   mkdir sticky && chmod +t,o-w sticky  && $test -k sticky &&
 	   mkdir owt    && chmod +t,o+w owt && $test -k owt &&
 	   mkdir owr    && chmod o+w owr)) == 0
 	     or (warn "$program_name: cannot create setuid/setgid/sticky files,"
